parquet-pig/src/main/java/org/apache/parquet/pig/convert/TupleConverter.java

```diff
@@ -530,6 +535,32 @@ public void addBinary(Binary value) {
   }
 
   /**
+   * handle decimal type
+   *
+   */
+  static final class FieldBigDecimalConverter extends PrimitiveConverter {
+    private final ParentValueContainer parent;
+    private final Type primitiveType;
+    public FieldBigDecimalConverter(Type primitiveType, ParentValueContainer parent) {
+      this.parent = parent;
+      this.primitiveType = primitiveType;
+    }
+
+    @Override
+    final public void addBinary(Binary value) {
+      if (primitiveType != null) {
+        int precision = primitiveType.asPrimitiveType().getDecimalMetadata().getPrecision();
+        int scale = primitiveType.asPrimitiveType().getDecimalMetadata().getScale();
+        BigDecimal finaldecimal = DecimalUtils.binaryToDecimal(value, precision, scale);
+        parent.add(finaldecimal);
+      } else {
+        parent.add("Broken DecimalType");
```

Comment:
I think I may just remove the check for primitiveType null... I was looking at the code and realized that there's already a check for it before newConverter gets called...
https://github.com/EllenKletscher/parquet-mr/blob/master/parquet-pig/src/main/java/org/apache/parquet/pig/convert/TupleConverter.java#L82